asset/js/widget/BaseInput.js

```diff
@@ -49,18 +49,23 @@ define(["../notjQuery", "Completer"], function ($, Completer) {
             $(this.input).on('input', this.onInput, this);
             $(this.input).on('keydown', this.onKeyDown, this);
             $(this.input).on('keyup', this.onKeyUp, this);
-            $(this.input).on('blur', this.onInputBlur, this);
             $(this.input).on('focusin', this.onTermFocus, this);
             $(this.termContainer).on('input', '[data-label]', this.onInput, this);
             $(this.termContainer).on('keydown', '[data-label]', this.onKeyDown, this);
-            $(this.termContainer).on('keyup', '[data-label]', this.onKeyUp, this);
-            $(this.termContainer).on('focusout', '[data-index]', this.onTermFocusOut, this);
-            $(this.termContainer).on('focusin', '[data-index]', this.onTermFocus, this);
+
+            if (! this.isReadOnlyMode()) {
```

Comment:
I've changed my mind here as well. `onTermFocusOut` makes really no sense to run in read-only mode, that's fine.

But term selection should still be possible (especially if the copy paste functionality is active), as this is not based on whether text can be selected. So `onInputBlur` and `onKeyUp` should always run. Only `onTermFocus` needs an update, it should still deselect terms, but the syntax check and suggestion retrieval is of no use, skip that in read only mode.